Tests/MongoSwiftTests/AsyncAwaitTestUtils.swift

```diff
@@ -0,0 +1,141 @@
+    internal func withTestClient<T>(
+        _ uri: String = MongoSwiftTestCase.getConnectionString().toString(),
+        options: MongoClientOptions? = nil,
+        eventLoopGroup: EventLoopGroup? = nil,
+        f: (MongoClient) async throws -> T
+    ) async throws -> T {
+        let elg = eventLoopGroup ?? MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(uri, eventLoopGroup: elg, options: options)
+        defer { client.syncCloseOrFail() }
+        return try await f(client)
+    }
+
+    // swiftlint:disable large_tuple
+    // see: https://github.com/realm/SwiftLint/issues/3753
+    internal func withTestNamespace<T>(
+        ns: MongoNamespace? = nil,
+        collectionOptions: CreateCollectionOptions? = nil,
+        _ f: (MongoClient, MongoDatabase, MongoCollection<BSONDocument>) async throws -> T
+    ) async throws -> T {
+        let ns = ns ?? self.getNamespace()
+        guard let collName = ns.collection else {
+            throw TestError(message: "missing collection")
+        }
+        return try await self.withTestClient { client in
+            let database = client.db(ns.db)
+            let collection: MongoCollection<BSONDocument>
+            do {
+                collection = try await database.createCollection(collName, options: collectionOptions)
+            } catch let error as MongoError.CommandError where error.code == 48 {
+                try await database.collection(collName).drop()
+                collection = try await database.createCollection(collName, options: collectionOptions)
+            }
+            defer { collection.syncDropOrFail() }
+            return try await f(client, database, collection)
+        }
+    }
+    // swiftlint:enable large_tuple
+}
+
+extension MongoClient {
+    internal func serverVersion() async throws -> ServerVersion {
+        let reply = try await self.db("admin").runCommand(
+            ["buildInfo": 1],
+            options: RunCommandOptions(readPreference: .primary)
+        )
+        guard let versionString = reply["version"]?.stringValue else {
+            throw TestError(message: " reply missing version string: \(reply)")
+        }
+        return try ServerVersion(versionString)
+    }
+
+    internal func serverParameters() async throws -> BSONDocument {
+        try await self.db("admin").runCommand(["getParameter": "*"])
+    }
+
+    internal func topologyType() async throws -> TestTopologyConfiguration {
+        async let helloReply = try self.db("admin").runCommand(["hello": 1])
+        async let shards = try self.db("config").collection("shards").find().get().toArray().get()
+        return try await TestTopologyConfiguration(helloReply: helloReply, shards: shards)
+    }
+
+    internal func getUnmetRequirement(_ testRequirement: TestRequirement) async throws -> UnmetRequirement? {
+        async let topologyType = try self.topologyType()
```

Comment:
just curious, is `async let` required to make these three evaluate simultaneously? Or would invoking them directly in the argument position of `getUnmetRequirements` behave the same way?

side note: GH doesn't seem able to highlight `async let` properly yet lol

Reply:
it does compile to one-line it like:
```swift
await testRequirement.getUnmetRequirement(givenCurrent: try self.serverVersion(), try self.topologyType(), try self.serverParameters())
```

but the behavior seems different - I roughly tested it by inserting print statements in each method before and after their `async` work like:
```swift
    internal func serverParameters() async throws -> BSONDocument {
        print("getting server parameters")
        let params = try await self.db("admin").runCommand(["getParameter": "*"])
        print("returning server parameters")
        return params
    }
```

as originally written, the interleaving of "getting x" and "returning x" varies on each run, for example one output is:
```
getting topology type
getting server version
getting server parameters
returning server version
returning topology type
returning server parameters
```

and another is
```
getting server version
getting topology type
getting server parameters
returning server parameters
returning server version
returning topology type
```

 which makes since given they are being run in parallel.

but written as you ask about in one line, the interleaving is always the same and suggests serial evaluation of each called function:
```
getting server version
returning server version
getting topology type
returning topology type
getting server parameters
returning server parameters
```